pkg/ensemble/iforest/forest.go

```diff
@@ -0,0 +1,175 @@
+package iforest
+
+import (
+	"math"
+	"math/rand"
+	"sync"
+)
+
+const (
+	defaultNumTrees       = 100
+	defaultSampleSize     = 256
+	defaultScoreThreshold = 0.6
+	defaultDetectionType  = DetectionTypeThreshold
+	offset                = 0.5
+)
+
+type DetectionType string
+
+const (
+	DetectionTypeThreshold  DetectionType = "threshold"
+	DetectionTypeProportion DetectionType = "proportion"
+)
+
+type Options struct {
+	DetectionType DetectionType `json:"detectionType"`
+	Threshold     float64       `json:"threshold"`
+	Proportion    float64       `json:"proportion"`
+	NumTrees      int           `json:"numTrees"`
+	SampleSize    int           `json:"sampleSize"`
+	MaxDepth      int           `json:"maxDepth"`
+}
+
+// SetDefaultValues initializes zero-valued fields in Options with defaults.
+func (o *Options) SetDefaultValues() {
+	if o.DetectionType == "" {
```

Comment:
The method name and comment do not indicate that it only sets default values when the value is not zero value. Should we remove all the if conditions?

Reply:
If we remove all the if conditions, then it will overrides the non-zero options in `NewWithOptions`.